content/blog/managing-google-workspace-with-terraform.md

```diff
@@ -0,0 +1,295 @@
+
+In terms of using the provider, while there are a few Terraform modules out there for managing Google Workspaces, we decided to create our own Terraform module. The motivation to build a new module from scratch stemmed from our desire to:
+
+1. Democratize user onboarding powers so Matt (CEO/CTO) is not a bottleneck.
+2. Provide transparency and history for changes to our team's permissions.
+3. Easily manage SSO configurations allowing engineers to sign into multiple AWS accounts using their Google Identity.
+4. Intuitively organize user and team settings in a clear and straight forward approach.
+
+Here's the GitHub link for the Google Workspace module:  
+[https://github.com/masterpointio/terraform-googleworkspace-users-groups-automation](https://github.com/masterpointio/terraform-googleworkspace-users-groups-automation)
+
+## Getting Started With the Module
+
+To make the module easy to get up and running with your own Google Workspace, we included two practical on-ramps for existing and brand new Google Workspaces:
+
+1. **Import an existing Google Workspace**  
+   We shared the [terraform code](https://github.com/masterpointio/terraform-googleworkspace-users-groups-automation/blob/main/examples/import-existing-org/main.tf) we used to declare existing [users](https://github.com/masterpointio/terraform-googleworkspace-users-groups-automation/blob/main/examples/import-existing-org/users.yaml) and [groups](https://github.com/masterpointio/terraform-googleworkspace-users-groups-automation/blob/main/examples/import-existing-org/groups.yaml) in YAML, and the [import blocks](https://github.com/masterpointio/terraform-googleworkspace-users-groups-automation/blob/main/examples/import-existing-org/imports.tf) to easily import an existing setup. We additionally shared a [Python script](https://github.com/masterpointio/terraform-googleworkspace-users-groups-automation/blob/main/examples/import-existing-org/debugging-script.py) to view the JSON of your existing resources (which you could give to an LLM and tell it to follow the example to create your own `users.yaml` and `groups.yaml` files)..
+
+2. **Create New Resources**  
+   Start fresh by managing Google Workspace users and groups directly from Terraform. This is a great starting point if you're setting up a new org or moving toward automation-first infra practices.
+
+After the initial import or setup, your Google Workspace [root module](https://newsletter.masterpoint.io/p/multi-instance-vs-single-instance-root-modules) could end up being as simple as this:
+
+```terraform
+locals {
+  oauth_scopes = [
+    "https://www.googleapis.com/auth/admin.directory.group",
+    "https://www.googleapis.com/auth/admin.directory.user",
+    "https://www.googleapis.com/auth/admin.directory.userschema",
+    "https://www.googleapis.com/auth/apps.groups.settings",
+    "https://www.googleapis.com/auth/iam",
+  ]
+}
+
+provider "googleworkspace" {
+  customer_id             = "my_customer"
```

Comment:
```suggestion
  customer_id             = "my_company"
```

isn't this example in the context of the user's company?

Reply:
Good call out. The `my_customer` is actually a specific setting for the underlying Google Workspace API and should remain `my_customer`. I'm sure others will have the same question. I'll add a comment briefly alluding to this (technical docs on github explicitly cover this).